File: PST/FUEL Source Code/FUEL/Backup/modCommonCode.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.modCommonCode
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System.Deployment.Application;
using System.Reflection;

namespace FUEL
{
  [StandardModule]
  internal sealed class modCommonCode
  {
    internal static string GetDataPath()
    {
      string str1;
      if (ApplicationDeployment.IsNetworkDeployed)
      {
        str1 = ApplicationDeployment.CurrentDeployment.DataDirectory.ToString();
      }
      else
      {
        string codeBase = Assembly.GetExecutingAssembly().CodeBase;
        string str2 = Strings.Left(codeBase, Strings.InStrRev(codeBase, "/"));
        str1 = Strings.Right(str2, checked (Strings.Len(str2) - Strings.InStr(str2, "///") - 2)).Replace("/", "\\");
      }
      return str1.EndsWith("\\") ? str1 : str1 + "\\";
    }
  }
}

File: PST/FUEL/src/FUEL/modCommonCode.vb
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Deployment.Application
Imports System.Reflection

Namespace FUEL
    <StandardModule> _
    Friend NotInheritable Class modCommonCode
        ' Methods
        Friend Shared Function GetDataPath() As String
            Dim codeBase As String
            If ApplicationDeployment.IsNetworkDeployed Then
                codeBase = ApplicationDeployment.CurrentDeployment.DataDirectory.ToString
            Else
                codeBase = Assembly.GetExecutingAssembly.CodeBase
                codeBase = Strings.Left(codeBase, Strings.InStrRev(codeBase, "/", -1, CompareMethod.Binary))
                codeBase = Strings.Right(codeBase, ((Strings.Len(codeBase) - Strings.InStr(codeBase, "///", CompareMethod.Binary)) - 2)).Replace("/", "\")
            End If
            Return If(Not codeBase.EndsWith("\"), (codeBase & "\"), codeBase)
        End Function

    End Class
End Namespace

